privacyIDEA-impl/src/main/resources/org/privacyidea/views/main.vm

```diff
@@ -71,9 +71,11 @@
                         </div>
 
                         <div class="form-element-wrapper">
-                            <input id="pi_otp_input" class="form-element form-field" name="pi_otp_input" type="password" autofocus placeholder=$piContext.getOtpFieldHint() value="">
+                            <input id="pi_otp_input" class="form-element form-field" name="pi_otp_input" type="password"
+                                   onKeyUp='if($piContext.getOtpLength()!=null&&this.value.length===$piContext.getOtpLength())document.getElementById("pi-form-submit-button").click()'
```

Comment:
1. please check if it is possible to extract the `onKeyUp` stuff to a function for readability
2. i think it is better to submit the form via `document.getElementById("privacyidea-form").submit()`

Reply:
1. I can try.
2. This submit button contains an additional event which is returned to the spring flow.

Reply:
👍

Reply:
[22ab4a9](https://github.com/privacyidea/shibboleth-plugin/pull/29/commits/22ab4a9afca649ee08b10c191a61a40672301536)